### ISSUE TITLE
List of HTTP codes to retry the petition should be customizable

### ISSUE THREAD

**Issue by sanacl** [chaoss/grimoirelab-perceval#578]
From grimoirelab-elk it is not possible to control when retries should be launched. In some cases like the one below, we need to retry when a 500 error response is provided by the server. Due to it is not possible to modify the hardcoded list of numeric HTTP responses, the data collection of Merge Requests from gitlab.com is taking more time than expected. 

```
Error feeding ocean from gitlab (https://gitlab.com/gitlab-org/gitlab-foss): 500 Server Error: Internal Server Error for url: https://gitlab.com/api/v4/projects/gitlab-org%2Fgitlab-foss/merge_requests/24937/notes?per_page=100&order_by=updated_at&sort=asc
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/grimoire_elk/elk.py", line 228, in feed_backend
    ocean_backend.feed(**params)
  File "/usr/local/lib/python3.5/dist-packages/grimoire_elk/raw/elastic.py", line 228, in feed
    self.feed_items(items)
  File "/usr/local/lib/python3.5/dist-packages/grimoire_elk/raw/elastic.py", line 244, in feed_items
    for item in items:
  File "/usr/local/lib/python3.5/dist-packages/perceval/backend.py", line 206, in fetch
    for item in self.fetch_items(category, **kwargs):
  File "/usr/local/lib/python3.5/dist-packages/perceval/backends/core/gitlab.py", line 295, in __fetch_merge_requests
    for mr_item in self.__fetch_merge_requests_data(fetch_from_date):
  File "/usr/local/lib/python3.5/dist-packages/perceval/backends/core/gitlab.py", line 335, in __fetch_merge_requests_data
    merge_full['notes_data'] = self.__get_merge_notes(merge_id)
  File "/usr/local/lib/python3.5/dist-packages/perceval/backends/core/gitlab.py", line 348, in __get_merge_notes
    for raw_notes in group_notes:
  File "/usr/local/lib/python3.5/dist-packages/perceval/backends/core/gitlab.py", line 635, in fetch_items
    response = self.fetch(url_next, payload=payload)
  File "/usr/local/lib/python3.5/dist-packages/perceval/backends/core/gitlab.py", line 619, in fetch
    response = super().fetch(url, payload, headers, method, stream)
  File "/usr/local/lib/python3.5/dist-packages/perceval/client.py", line 132, in fetch
    response = self._fetch_from_remote(url, payload, headers, method, stream, verify, auth)
  File "/usr/local/lib/python3.5/dist-packages/perceval/client.py", line 173, in _fetch_from_remote
    raise e
  File "/usr/local/lib/python3.5/dist-packages/perceval/client.py", line 168, in _fetch_from_remote
    response.raise_for_status()
  File "/usr/local/lib/python3.5/dist-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://gitlab.com/api/v4/projects/gitlab-org%2Fgitlab-foss/merge_requests/24937/notes?per_page=100&order_by=updated_at&sort=asc
```

The HTTP responses where Perceval retry to get content should be customizable.

**Comment by valeriocos** [comment]
A PR addresses this issue: https://github.com/chaoss/grimoirelab-perceval/pull/579

please @zhquan @sanacl have a look at it

**Comment by valeriocos** [comment]
Closed by #579
